From c4576332b7e93aa81d7af1284bac0d1e6b13cb85 Mon Sep 17 00:00:00 2001
From: Panu Matilainen <pmatilai@redhat.com>
Date: Tue, 18 Feb 2014 12:49:39 +0200
Subject: [PATCH] Warn on empty %files -f manifest files

- While there's nothing wrong with packages with no files, a manifest
  file with no files is likely to be an error or mistake of some
  sort and seems worth issuing a warning at least.
- Related to / inspired by ticket #870.

(cherry picked from commit e7a60880931743a7ac0c7e2765d447f21ec18a7c)
Signed-off-by: Quentin Casasnovas <quentin.casasnovas@vates.tech>
---
 build/files.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/build/files.c b/build/files.c
index d8fe5384aa67..c26d9d91e1fe 100644
--- a/build/files.c
+++ b/build/files.c
@@ -1648,6 +1648,7 @@ static rpmRC readFilesManifest(rpmSpec spec, Package pkg, const char *path)
     char *fn, buf[BUFSIZ];
     FILE *fd = NULL;
     rpmRC rc = RPMRC_FAIL;
+    unsigned int nlines = 0;
 
     if (*path == '/') {
 	fn = rpmGetPath(path, NULL);
@@ -1669,8 +1670,12 @@ static rpmRC readFilesManifest(rpmSpec spec, Package pkg, const char *path)
 	    goto exit;
 	}
 	argvAdd(&(pkg->fileList), buf);
+	nlines++;
     }
 
+    if (nlines == 0)
+	rpmlog(RPMLOG_WARNING, _("Empty %%files file %s\n"), fn);
+
     if (ferror(fd))
 	rpmlog(RPMLOG_ERR, _("Error reading %%files file %s: %m\n"), fn);
     else
